shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -107,3 +130,133 @@
 }
 
 }  // namespace shell
+
+// This recognizers delays touch events from being dispatched to the responder chain until it failed
+// recognizing a gesture.
+//
+// We only fail this recognizer when asked to do so by the Flutter framework (which does so by
+// invoking an acceptGesture method on the platform_views channel). And this is how we allow the
+// Flutter framework to delay or prevent the embedded view from getting a touch sequence.
+@interface DelayingGestureRecognizer : UIGestureRecognizer <UIGestureRecognizerDelegate>
+@end
+
+// While the DelayingGestureRecognizer is preventing touches from hitting the responder chain
+// the touch events are not arriving to the FlutterView (and thus not arriving to the Flutter
+// framework). We use this gesture recognizer to dispatch the events directly to the FlutterView
+// while during this phase.
+//
+// If the Flutter framework decides to dispatch events to the embedded view, we fail the
+// DelayingGestureRecognizer which sends the events up the responder chain. But since the events
+// are handled by the embedded view they are not delivered to the Flutter framework in this phase
+// as well. So during this phase as well the ForwardingGestureRecognizer dispatched the events
+// directly to the FlutterView.
+@interface ForwardingGestureRecognizer : UIGestureRecognizer <UIGestureRecognizerDelegate>
+- (instancetype)initWithTarget:(id)target flutterView:(UIView*)flutterView;
+@end
+
+@implementation TouchInterceptingView {
+  ForwardingGestureRecognizer* forwardingRecognizer;
+  DelayingGestureRecognizer* delayingRecognizer;
+}
+- (instancetype)initWithSubView:(UIView*)embeddedView flutterView:(UIView*)flutterView {
+  self = [super initWithFrame:embeddedView.frame];
+  if (self) {
+    self.multipleTouchEnabled = YES;
+    embeddedView.autoresizingMask =
+        (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
+
+    [self addSubview:embeddedView];
+
+    forwardingRecognizer =
+        [[[ForwardingGestureRecognizer alloc] initWithTarget:self
+                                                 flutterView:flutterView] autorelease];
+
+    delayingRecognizer = [[[DelayingGestureRecognizer alloc] initWithTarget:self
```

Comment:
What is the point of these ivars at all? You don't end up using the same.

Reply:
In any case, I think it would be better to hold a strong reference to the recognizer.

Reply:
the delaying recognizer is used when releaseGesture is called (which is triggered by the framework).

I removed the forwardingrecognizer ivar reference.